streams/src/test/java/org/apache/kafka/streams/processor/internals/AbstractTaskTest.java

```diff
@@ -105,24 +119,123 @@ public void shouldNotAttemptToLockIfNoStores() throws IOException {
         EasyMock.verify(stateDirectory);
     }
 
-    private AbstractTask createTask(final Consumer consumer, final List<StateStore> stateStores) {
+    @Test
+    public void shouldDeleteAndRecreateStoreDirectoryOnReinitialize() throws IOException {
+        final StreamsConfig streamsConfig = new StreamsConfig(new Properties() {
+            {
+                put(StreamsConfig.APPLICATION_ID_CONFIG, "app-id");
+                put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+                put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getAbsolutePath());
+            }
+        });
+        final Consumer consumer = EasyMock.createNiceMock(Consumer.class);
+
+        final StateStore store1 = EasyMock.createNiceMock(StateStore.class);
```

Comment:
Do we need to do this for so many stores? i think 2 would be sufficient as the test becomes quite noisy and it is difficult to see the intent

Reply:
We have 4 cases we need to check: with RocksDB and w/o RocksDB, and for both cases we need to make sure only the affected state directory is delete but the a second (non-affected) state directly. We could split the test in two -- but it would duplicate quite some code. Comments?